src/Symfony/Component/Security/Http/Firewall/ContextListener.php

```diff
@@ -168,6 +168,15 @@ protected function refreshUser(TokenInterface $token)
                 $refreshedUser = $provider->refreshUser($user);
                 $token->setUser($refreshedUser);
 
+                // tokens can be deauthenticated if the user has been changed.
+                if (!$token->isAuthenticated()) {
+                    if (null !== $this->logger) {
+                        $this->logger->debug('Token was deauthenticated after trying to refresh it.', array('username' => $refreshedUser->getUsername(), 'provider' => get_class($provider)));
+                    }
+
+                    @trigger_error('Refreshing a deauthenticated user is deprecated as of 3.4 and will trigger a logout in 4.0.', E_USER_DEPRECATED);
```

Comment:
I will add them as soon as I know how to proceed with the actual logging out users, as this might need more additions to the changelog and upgrade entries 👍